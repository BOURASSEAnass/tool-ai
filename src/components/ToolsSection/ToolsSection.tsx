import { tools } from "../../data/tools";
import Container from "../Container/Container";
import ToolCardList from "../ToolCardList/ToolCardList";

const ToolsSection = () => {
  return (
    <section className="flex flex-col items-center bg-tertiary py-10">
      <Container>
        <div className="flex flex-col gap-4">
          <h2 className="text-4xl font-semibold text-background">Tools</h2>
          <ToolCardList tools={tools} />
        </div>
      </Container>
    </section>
  );
};

export default ToolsSection;
