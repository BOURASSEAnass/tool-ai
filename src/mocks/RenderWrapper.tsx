import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { store } from "../redux/store";

interface RenderWrapperProps {
  children: JSX.Element | JSX.Element[];
}

const RenderWrapper = ({ children }: RenderWrapperProps) => {
  return (
    <BrowserRouter>
      <Provider store={store}>{children}</Provider>
    </BrowserRouter>
  );
};

export default RenderWrapper;
