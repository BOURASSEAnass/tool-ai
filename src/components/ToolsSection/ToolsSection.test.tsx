import { render, screen } from "@testing-library/react";
import RenderWrapper from "../../mocks/RenderWrapper";

import ToolsSection from "./ToolsSection";

describe("Given a ToolsSection component", () => {
  describe("When it's rendered", () => {
    test("Then it should match the snapshot", () => {
      const view = render(
        <RenderWrapper>
          <ToolsSection />
        </RenderWrapper>
      );

      expect(view).toMatchSnapshot();
    });

    test("Then it should have a heading level 2 'Tools'", () => {
      const expectedHeadingName = "Tools";

      render(
        <RenderWrapper>
          <ToolsSection />
        </RenderWrapper>
      );

      const heading = screen.getByRole("heading", {
        name: expectedHeadingName,
      });

      expect(heading).toBeInTheDocument();
    });
  });
});
