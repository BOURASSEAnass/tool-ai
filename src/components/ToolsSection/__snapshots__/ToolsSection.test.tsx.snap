// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ToolsSection component When it's rendered Then it should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="flex flex-col items-center bg-tertiary py-10"
      >
        <div
          class="container mx-auto max-w-6xl flex flex-col p-4"
        >
          <div
            class="flex flex-col gap-4"
          >
            <h2
              class="text-4xl font-semibold text-background"
            >
              Tools
            </h2>
            <ul
              class="flex flex-col gap-12 md:flex-row md:grid grid-cols-3"
            >
              <li>
                <article
                  class="w-full flex flex-col items-center gap-2 bg-primary max-w-sm text-background p-2 border-2 border-background rounded-sm shadow-lg"
                >
                  <h3
                    class="font-bold text-3xl"
                  >
                    Spam
                  </h3>
                  <p
                    class="text-center"
                  >
                    Check if a email is spam with one click
                  </p>
                  <a
                    class="border-2 border-background rounded-md py-0.5 px-7 font-semibold hover:bg-tertiary"
                    href="/spam"
                  >
                    Try it
                  </a>
                </article>
              </li>
              <li>
                <article
                  class="w-full flex flex-col items-center gap-2 bg-primary max-w-sm text-background p-2 border-2 border-background rounded-sm shadow-lg"
                >
                  <h3
                    class="font-bold text-3xl"
                  >
                    Spam
                  </h3>
                  <p
                    class="text-center"
                  >
                    Check if a email is spam with one click
                  </p>
                  <a
                    class="border-2 border-background rounded-md py-0.5 px-7 font-semibold hover:bg-tertiary"
                    href="/spam"
                  >
                    Try it
                  </a>
                </article>
              </li>
              <li>
                <article
                  class="w-full flex flex-col items-center gap-2 bg-primary max-w-sm text-background p-2 border-2 border-background rounded-sm shadow-lg"
                >
                  <h3
                    class="font-bold text-3xl"
                  >
                    Spam
                  </h3>
                  <p
                    class="text-center"
                  >
                    Check if a email is spam with one click
                  </p>
                  <a
                    class="border-2 border-background rounded-md py-0.5 px-7 font-semibold hover:bg-tertiary"
                    href="/spam"
                  >
                    Try it
                  </a>
                </article>
              </li>
              <li>
                <article
                  class="w-full flex flex-col items-center gap-2 bg-primary max-w-sm text-background p-2 border-2 border-background rounded-sm shadow-lg"
                >
                  <h3
                    class="font-bold text-3xl"
                  >
                    Spam
                  </h3>
                  <p
                    class="text-center"
                  >
                    Check if a email is spam with one click
                  </p>
                  <a
                    class="border-2 border-background rounded-md py-0.5 px-7 font-semibold hover:bg-tertiary"
                    href="/spam"
                  >
                    Try it
                  </a>
                </article>
              </li>
            </ul>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="flex flex-col items-center bg-tertiary py-10"
    >
      <div
        class="container mx-auto max-w-6xl flex flex-col p-4"
      >
        <div
          class="flex flex-col gap-4"
        >
          <h2
            class="text-4xl font-semibold text-background"
          >
            Tools
          </h2>
          <ul
            class="flex flex-col gap-12 md:flex-row md:grid grid-cols-3"
          >
            <li>
              <article
                class="w-full flex flex-col items-center gap-2 bg-primary max-w-sm text-background p-2 border-2 border-background rounded-sm shadow-lg"
              >
                <h3
                  class="font-bold text-3xl"
                >
                  Spam
                </h3>
                <p
                  class="text-center"
                >
                  Check if a email is spam with one click
                </p>
                <a
                  class="border-2 border-background rounded-md py-0.5 px-7 font-semibold hover:bg-tertiary"
                  href="/spam"
                >
                  Try it
                </a>
              </article>
            </li>
            <li>
              <article
                class="w-full flex flex-col items-center gap-2 bg-primary max-w-sm text-background p-2 border-2 border-background rounded-sm shadow-lg"
              >
                <h3
                  class="font-bold text-3xl"
                >
                  Spam
                </h3>
                <p
                  class="text-center"
                >
                  Check if a email is spam with one click
                </p>
                <a
                  class="border-2 border-background rounded-md py-0.5 px-7 font-semibold hover:bg-tertiary"
                  href="/spam"
                >
                  Try it
                </a>
              </article>
            </li>
            <li>
              <article
                class="w-full flex flex-col items-center gap-2 bg-primary max-w-sm text-background p-2 border-2 border-background rounded-sm shadow-lg"
              >
                <h3
                  class="font-bold text-3xl"
                >
                  Spam
                </h3>
                <p
                  class="text-center"
                >
                  Check if a email is spam with one click
                </p>
                <a
                  class="border-2 border-background rounded-md py-0.5 px-7 font-semibold hover:bg-tertiary"
                  href="/spam"
                >
                  Try it
                </a>
              </article>
            </li>
            <li>
              <article
                class="w-full flex flex-col items-center gap-2 bg-primary max-w-sm text-background p-2 border-2 border-background rounded-sm shadow-lg"
              >
                <h3
                  class="font-bold text-3xl"
                >
                  Spam
                </h3>
                <p
                  class="text-center"
                >
                  Check if a email is spam with one click
                </p>
                <a
                  class="border-2 border-background rounded-md py-0.5 px-7 font-semibold hover:bg-tertiary"
                  href="/spam"
                >
                  Try it
                </a>
              </article>
            </li>
          </ul>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
